libs/counterfactuals/src/lib/CounterfactualComponent.tsx

```diff
@@ -53,6 +54,7 @@ export interface ICounterfactualComponentState {
   indexSeries: number[];
   isCounterfactualsDataLoading?: boolean;
   localCounterfactualErrorMessage?: string;
+  isRevertButtonClicked: boolean;
```

Comment:
I'm slightly confused why a button click has an entire state variable representing it. Is there some async call that happens after the button click that we need to wait for?

If not, is it possible to collapse the changes happening to set and unset this flag into a single set of updates (thus rendering this state variable unnecessary)?

Reply:
I need this variable to propagate to LargeCounterfactualChart to update bubble chart. I am happy to update if there is any other recommended way to handle this.

Reply:
I see. And LargeCounterfactualChart is within this one. I guess the pattern I've seen for this type of situation is to have the state at a higher level (e.g., in this file) and pass it in. Can't say I like that a lot either since a lot of state can get concentrated at higher levels, but at least the state update happens in 1 step rather than 2.